lib/formdata.c

```diff
@@ -627,9 +573,14 @@ CURLFORMcode FormAdd(struct curl_httppost **httppost,
          !form->contenttype) {
         char *f = form->flags & HTTPPOST_BUFFER?
           form->showfilename : form->value;
+        char const *type = prevtype;
+        if(!type)
```

Comment:
To stay compatible, prevtype should be used only if Curl_mime_contenttype(f) return NULL.

Reply:
Oops, you're right. Thanks. I'll fix!